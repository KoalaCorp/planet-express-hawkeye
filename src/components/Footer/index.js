import React from "react"
import { Link } from "react-router-dom"

import Styled from "./styled"

const Footer = () => {
  return (
    <Styled.Footer>
      <Styled.Navigation>
        <li>
          <Link to="/about">about the project</Link>
        </li>
        <li>
          <Link to="/team">about the team</Link>
        </li>
      </Styled.Navigation>

      <Styled.License>
        this project is under a{" "}
        <a
          href="https://www.gnu.org/licenses/gpl-3.0.en.html"
          title="GNU General Public License v3.0"
          rel="license external noopener noreferrer"
        >
          GNU General Public License v3.0
        </a>{" "}
        license
      </Styled.License>
    </Styled.Footer>
  )
}

export default Footer
