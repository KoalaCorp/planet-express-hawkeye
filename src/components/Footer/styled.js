import styled from "styled-components"

import tools from "../../assets/styles/tools"

export default {
  Footer: styled.footer`
    text-align: center;
    background: ${(props) => props.theme.settings.colors.palette.tertiary};
    padding: 2rem;
    min-height: 6rem;

    @media ${tools.devices.lg} {
      text-align: left;
      display: flex;
      align-items: center;
    }
  `,

  Navigation: styled.ul`
    list-style: none;

    @media ${tools.devices.lg} {
      flex: 1;
    }

    li {
      display: inline;
      margin-right: 2rem;
    }
  `,

  License: styled.div`
    margin-top: 2rem;

    @media ${tools.devices.lg} {
      margin-top: 0;
    }
  `,
}
