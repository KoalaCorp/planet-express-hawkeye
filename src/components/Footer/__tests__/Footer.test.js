import React from "react"
import { render } from "../../../config/tests/helpers"

import Footer from "../"

const renderComponent = ({ ...props } = {}) => render(<Footer {...props} />)

describe("Footer", () => {
  it("renders correctly", () => {
    const footer = renderComponent()
    expect(footer).toMatchSnapshot()
  })
})
