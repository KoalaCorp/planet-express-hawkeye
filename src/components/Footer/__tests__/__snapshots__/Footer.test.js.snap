// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="styled__Footer-sc-1fwt48x-0 bBFSxL"
      >
        <ul
          class="styled__Navigation-sc-1fwt48x-1 kvXOAi"
        >
          <li>
            <a
              href="/about"
            >
              about the project
            </a>
          </li>
          <li>
            <a
              href="/team"
            >
              about the team
            </a>
          </li>
        </ul>
        <div
          class="styled__License-sc-1fwt48x-2 kXCNav"
        >
          this project is under a
           
          <a
            href="https://www.gnu.org/licenses/gpl-3.0.en.html"
            rel="license external noopener noreferrer"
            title="GNU General Public License v3.0"
          >
            GNU General Public License v3.0
          </a>
           
          license
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="styled__Footer-sc-1fwt48x-0 bBFSxL"
    >
      <ul
        class="styled__Navigation-sc-1fwt48x-1 kvXOAi"
      >
        <li>
          <a
            href="/about"
          >
            about the project
          </a>
        </li>
        <li>
          <a
            href="/team"
          >
            about the team
          </a>
        </li>
      </ul>
      <div
        class="styled__License-sc-1fwt48x-2 kXCNav"
      >
        this project is under a
         
        <a
          href="https://www.gnu.org/licenses/gpl-3.0.en.html"
          rel="license external noopener noreferrer"
          title="GNU General Public License v3.0"
        >
          GNU General Public License v3.0
        </a>
         
        license
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
